packages/react/src/hooks/useTimeout.ts

```diff
@@ -1,15 +1,11 @@
-import { useEffect, useRef } from 'react'
-import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect'
+import { useEffect } from 'react'
+import { usePreservedCallback } from './usePreservedCallback'
 
 export const useTimeout = (fn: () => void, ms: number) => {
-  const fnRef = useRef(fn)
-
-  useIsomorphicLayoutEffect(() => {
-    fnRef.current = fn
-  }, [fn])
+  const preservedCallback = usePreservedCallback(fn)
```

Comment:
I think Components([Delay](https://github.com/toss/suspensive/blob/main/packages/react/src/Delay.tsx#L21)) using useTimeout should use useCallback too.

Reply:
@ssi02014 I absolutely accept your opinion https://github.com/toss/suspensive/pull/966#issuecomment-2185931597 but anyway

I think Components([Delay](https://github.com/toss/suspensive/blob/main/packages/react/src/Delay.tsx#L21)) using useTimeout should use useCallback too. doesn't it right?

### TO-BE
```diff
import { type PropsWithChildren, type ReactNode, useContext, useState } from 'react'
import { DelayDefaultPropsContext } from './contexts'
import { useTimeout } from './hooks'
import { Message_Delay_ms_prop_should_be_greater_than_or_equal_to_0, SuspensiveError } from './models/SuspensiveError'

export interface DelayProps extends PropsWithChildren {
  ms?: number
  fallback?: ReactNode
}

export const Delay = (props: DelayProps) => {
  if (process.env.NODE_ENV === 'development') {
    if (typeof props.ms === 'number') {
      SuspensiveError.assert(props.ms >= 0, Message_Delay_ms_prop_should_be_greater_than_or_equal_to_0)
    }
  }
  const defaultProps = useContext(DelayDefaultPropsContext)
  const ms = props.ms ?? defaultProps.ms ?? 0

  const [isDelaying, setIsDelaying] = useState(ms > 0)
-  useTimeout(() => setIsDelaying(false), ms)
+  useTimeout(useCallback(() => setIsDelaying(false), []), ms)

  const fallback = typeof props.fallback === 'undefined' ? defaultProps.fallback : props.fallback
  return <>{isDelaying ? fallback : props.children}</>
}
if (process.env.NODE_ENV === 'development') {
  Delay.displayName = 'Delay'
}
```

Reply:
@manudeli Oh no, `useTimeout` works the same way.
No need to change it! 👍

Reply:
Ah! I understood what you mean! Thanks